packages/api/src/resolvers/article/index.ts

```diff
@@ -619,13 +619,43 @@ export const saveArticleReadingProgressResolver = authorized<
     ) {
       return { errorCodes: [SaveArticleReadingProgressErrorCode.BadData] }
     }
-    try {
+
+    // We don't need to update the values of reading progress here
+    // because the function resolver will handle that for us when
+    // it resolves the properties of the Article object
+    let updatedItem = await findLibraryItemById(id, uid)
+    if (!updatedItem) {
+      return {
+        errorCodes: [SaveArticleReadingProgressErrorCode.Unauthorized],
+      }
+    }
+
+    if (env.redis.cache && env.redis.mq) {
```

Comment:
We don't have to right now, but we can probably remove all the `if (redis)` type checks in backend now and make it mandatory, since we are simplifying so much of self hosting with it.